### PR TITLE
General style cleanup (header, data catalog, applications)

### PR DESCRIPTION
**Pull request by jfrankl**
### Header
- Right align "Request access" button
- Enhance vertical padding
- Minor enhancements to make the overall size and spacing more closely match other Microsoft websites
Screen Shot 2021-12-30 at 3 39 20 PM

### Data catalog
- Right-align the last column in a row with the right side of the page
- Move the tag filter to be closer to the content it filters
- Visually distinguish tags from buttons
- Minor spacing enhancements
Screen Shot 2021-12-30 at 5 07 19 PM

### Applications
- Remove margin from headings so the main content aligns with the main content on the "Data catalog" page
Screen Shot 2021-12-30 at 3 35 08 PM

### Dataset Detail
- Use headings instead of text for metadata
- Remove padding on bottom of sticky header, which was causing a gap
Screen Shot 2021-12-31 at 3 43 42 PM

### CODE REVIEW

**Review by alkamin — APPROVED** [review]
Code-wise this looks good to me.

File: src/styles/catalog.css [hunk 1]
```diff
@@ -4,7 +4,7 @@
 .layout-row {
   display: flex;
   flex-flow: row wrap;
-  justify-content: flex-start;
+  justify-content: space-between;
```
Comment (mmcfarland) [inline src/styles/catalog.css:7]: I think this change alters the layout of the "additional datasets" section when there are few results. Compare the [existing staging](https://planetarycomputer-staging.microsoft.com/catalog?tags=Water) to [this branch](https://icy-meadow-0fc35e30f-211.eastus2.azurestaticapps.net/catalog?tags=Water)

image

**Review by mmcfarland — COMMENTED** [review]
Thanks @jfrankl, this tightens things up nicely. Besides the one spacing comment below, another thing to consider is the placement of the tag filter below the API dataset heading. Technically, that filter applies to that API section and the section below ("additional datasets"), which is why it was above both. In practical terms, a user would rarely know _which_ section their filter will give the results they want, so they're likely to look at both. The top (main) section is also our more prominent data, which is why it's displayed so much more prominently. I'm not against the new location, since it does have a nicer flow, but just wanted to point out the behavior in case you missed it.

image

File: src/styles/catalog.css (hunk 1 above)
Comment (jfrankl) [reply src/styles/catalog.css:7]: Great catch.

**Comment by mmcfarland** [comment]
Closing in favor of #224 which resolve commit history issues.

**Pull request closed without merging**